### PR TITLE
OpenBSD doc

### PR DESCRIPTION
**Pull request by huskyii**
- [x] read the [CONTRIBUTING guidelines](README.md#contributing)
- [ ] raised a GitHub issue or discussed it on the projects chat beforehand
- [ ] added unit tests
- [ ] added integration tests
- [x] updated documentation if needed
- [ ] updated CHANGELOG.md

### CODE REVIEW

**Comment by e-zk** [comment]
As an OpenBSD user myself I can confirm these instructions are good :)

Pretty similar setup to [a blog post](https://zakaria.org/posts/headscale-setup.html) I wrote a while ago.

File: docs/running-headscale-openbsd.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+# 2. gmake: Makefile in the headscale repo is written in GNU make syntax
+pkg_add -D snap go
+pkg_add gmake
+
+git clone https://github.com/juanfont/headscale.git
+
+cd headscale
+
+# optionally checkout a release
+git checkout v0.16.0-beta1
```
Comment (kradalby) [inline docs/running-headscale-openbsd.md:30]: Maybe we should have a note here to check the releases so we dont hardcode a version?

**Review by kradalby — COMMENTED** [review]
Looks sensible, just one small nit

File: docs/running-headscale-openbsd.md (hunk 1 above)
Comment (huskyii) [reply docs/running-headscale-openbsd.md]: Good suggestion :), updated.

File: docs/running-headscale-openbsd.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+# Install prerequistes
+# 1. go v1.18+: headscale newer than 0.15 needs go 1.18+ to compile
+# 2. gmake: Makefile in the headscale repo is written in GNU make syntax
+
+git clone https://github.com/juanfont/headscale.git
+
+cd headscale
+
+# optionally checkout a release
+git checkout v0.16.0-beta1
```
Comment (kradalby) [inline docs/running-headscale-openbsd.md:57]: There is one more :)

Comment (huskyii) [reply docs/running-headscale-openbsd.md:57]: I need refactor the doc to DRY here :(

Comment (huskyii) [reply docs/running-headscale-openbsd.md]: Can't DRY with plain markdown -.-

**Review by kradalby — APPROVED** [review]

**Pull request merged**